src/openrct2/GameState.cpp

```diff
@@ -0,0 +1,252 @@
+#pragma region Copyright (c) 2014-2017 OpenRCT2 Developers
+/*****************************************************************************
+* OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+*
+* OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+* For more information, visit https://github.com/OpenRCT2/OpenRCT2
+*
+* OpenRCT2 is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* A full copy of the GNU General Public License can be found in licence.txt
+*****************************************************************************/
+#pragma endregion
+
+#include "GameState.h"
+#include "Context.h"
+#include "core/Math.hpp"
+#include "Editor.h"
+#include "Input.h"
+#include "interface/Screenshot.h"
+#include "localisation/Date.h"
+#include "management/NewsItem.h"
+#include "network/network.h"
+#include "OpenRCT2.h"
+#include "platform/Platform2.h"
+#include "scenario/Scenario.h"
+#include "title/TitleScreen.h"
+#include "title/TitleSequencePlayer.h"
+#include "world/Climate.h"
+#include "world/MapAnimation.h"
+#include "world/Park.h"
+
+using namespace OpenRCT2;
+
+GameState::GameState()
+{
+    _park = std::make_unique<Park>();
```

Comment:
This appears to be deleted soon as it's set. The game crashes accordingly:
```
(lldb) run
Process 3960 launched: './openrct2' (x86_64)
Process 3960 stopped
* thread #1, name = 'openrct2', stop reason = signal SIGSEGV: invalid address (fault address: 0x0)
    frame #0: 0x00005555558bda05 openrct2`std::__uniq_ptr_impl<OpenRCT2::Park, std::default_delete<OpenRCT2::Park> >::_M_ptr() const + 21
openrct2`std::__uniq_ptr_impl<OpenRCT2::Park, std::default_delete<OpenRCT2::Park> >::_M_ptr:
->  0x5555558bda05 <+21>: movq   (%rax), %rax
    0x5555558bda08 <+24>: addq   $0x10, %rsp
    0x5555558bda0c <+28>: popq   %rbp
    0x5555558bda0d <+29>: retq
```

Reply:
```
>	openrct2.dll!OpenRCT2::Context::GetGameState() Line 172	C++
 	openrct2.dll!park_init() Line 1065	C++
 	openrct2.dll!game_init_all(int mapSize=150) Line 1477	C++
 	openrct2.dll!OpenRCT2::Context::Initialise() Line 454	C++
```

The problem is that `park_init` is calling GetGameState before the game state is created, in fact it is the creation of game state that calls `park_init`.

Reply:
I've done the latter, but it seems I've missed something, still. Could you have another look?